installer/install.iss

```diff
@@ -974,9 +979,13 @@ begin
         i:=InferredDefaultKeys.IndexOf(Key); // cannot use .Find because the list is not sorted
         if (i>=0) then
             Result:=InferredDefaultValues[i]
-        else
+        else begin
             // Restore the settings chosen during a previous install.
             Result:=GetPreviousData(Key,Default);
+            // Test to see if we really set this to the Default value
+            if ('z'+Default=GetPreviousData(Key,'z'+Default)) then
```

Comment:
If -- for whatever abstruse reason -- the previously recorded choice happens to be identical to `'z'+Default`, this condition does not do what we want.

I really think that we need a different condition here. Let's see. We need to call `GetPreviousData()` again, that's for sure. We cannot change the `Key` parameter, because we want to see whether there has been any pre-recorded value. So we have to play with the second parameter, i.e. with the default value to use if no value has been recorded in previous installs. Therefore we need to use a default value there that must be different from whatever the `GetPreviousData()` call in line 984 could possibly have returned. So if we modify the value that was actually returned, e.g. `'z'+Result`, we have _exactly_ what we need.

Therefore, I think the condition you want is actually:

```pascal
if ('z'+Result=GetPreviousData(Key,'z'+Result)) then
```

Now, I still think that it would be much easier to reason about if we looked only for the pages that we want to show, rather than for the options that we do not need to show because the user saw them already.

In other words, I would suggest something like this instead:

```pascal
if (Pos(','+PrevPageID+',',PagesWithPreviouslyUnseenOptions)=0) then begin
    if (PagesWithPreviouslyUnseenOptions='') then
        PagesWithPreviouslyUnseenOptions:=',';
    PagesWithPreviouslyUnseenOptions:=PagesWithPreviouslyUnseenOptions+IntToStr(PrevPageID)+',';
end
```

Reply:
The `'z'+Result` fix is a good idea; I will change that.